### PR TITLE
Y axis labels for value plots

### LINKED ISSUE #261: Axis values not showing up on value plots properly
Depending on what you set your range to, the numbers along the plot's vertical axis won't show up. For example, if you set the range on a value plot to "-2,2" these numbers will show up, but if you set it to "-3,3" then no numbers on the ends will show up.

### PR DESCRIPTION
**Pull request by cchan987**
Instead of letting d3 automatically generate ticks as it want, Enforce that it labels the min and max values of the value plot range.

Fix #261

### CODE REVIEW

File: nengo_gui/static/2d_axes.js [hunk 1]
```diff
@@ -38,9 +38,12 @@ Nengo.Axes2D = function(parent, args) {
         .scale(this.scale_y)
         .orient("left")
         .ticks(2)
+        .tickValues([-1,0,1])
```
Comment (tcstewar) [inline nengo_gui/static/2d_axes.js]: Hmm, shouldn't this be ```[args.min_value, args.max_value]```?  And isn't the ```ticks(2)``` redundant?

File: nengo_gui/static/value.js [hunk 2]
```diff
@@ -148,6 +150,7 @@ Nengo.Value.prototype.set_range = function() {
             var max = parseFloat(new_range[1]);
             self.update_range(min, max);
             self.save_layout();
+            self.axes2d.axis_y.tickValues([min,max])
```
Comment (tcstewar) [inline nengo_gui/static/value.js]: spacing after the comma....  :)

**Comment by cchan987** [comment]
updated

**Comment by tcstewar** [comment]
Hmm...  when the graphs render for me (Windows, Chrome and Firefox), I get the "minus" sign disappearing from the bottom one (due to it rendering as "-1.0" rather than "-1".  This means that the graphs look like they go from "1.0" to "1.0", which is a bit of a problem.  So this change is making the bug in #450 much worse (since it now happens for pretty much all graphs, rather than just ones where people have changed the range).

Not sure what the right solution is, other than to modify this one somehow so that it would end up displaying the axis labels as "1" and "-1" rather than "1.0" and "-1.0"....

**Comment by cchan987** [comment]
I'll probably merge this and #450 into one PR. Feels like they're related enough that we can get 2 birds with 1 stone here.

**Comment by cchan987** [comment]
This PR also fixes #450 now

**Comment by tcstewar** [comment]
Nice!  That looks like a pretty good fix to me....

File: nengo_gui/static/2d_axes.js [hunk 3]
```diff
@@ -85,3 +87,21 @@ Nengo.Axes2D.prototype.on_resize = function(width, height) {
...
 };
 
+Nengo.Axes2D.prototype.fit_ticks = function(parent){
+    var self = this;
+    setTimeout(function(){
+        var ticks = $(parent.div).find('.tick');
+        var max_w = 0;
+        for (var i = 0; i < ticks.length; i++) {
+            var w = ticks[i].getBoundingClientRect().width
+            console.log(w)
```
Comment (tcstewar) [inline nengo_gui/static/2d_axes.js]: This log should probably be removed

File: nengo_gui/static/2d_axes.js [hunk 4]
```diff
@@ -85,3 +87,21 @@ Nengo.Axes2D.prototype.on_resize = function(width, height) {
     this.axis_y_g.call(this.axis_y);
 };
 
+Nengo.Axes2D.prototype.fit_ticks = function(parent){
+    var self = this;
+    setTimeout(function(){
+        var ticks = $(parent.div).find('.tick');
+        var max_w = 0;
+        for (var i = 0; i < ticks.length; i++) {
+            var w = ticks[i].getBoundingClientRect().width
```
Comment (tcstewar) [inline nengo_gui/static/2d_axes.js]: semicolon missing....

**Comment by tcstewar** [comment]
Seems okay to me.  :)  merging...

**Pull request merged**